pkg/costmodel/costmodel.go

```diff
@@ -1158,6 +1158,7 @@ func (cm *CostModel) GetNodeCost(cp costAnalyzerCloud.Provider) (map[string]*cos
 		if newCnode.Region == "" {
 			newCnode.Region = n.Labels[v1.LabelZoneRegion]
 		}
+		newCnode.ProviderID = n.Spec.ProviderID
```

Comment:
On-prem clusters, and other custom provisioners, may not have this field set. Just want to make sure that scenario wouldn't cause problems for you here.

Reply:
Shouldn't be, that field would just be an empty string, which seems appropriate.